Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,533 @@
+            } else {
+                var sinPhi = Math.sin(start.latitude);
+                localRad = major * Math.cos(start.latitude) / Math.sqrt(1 - ellipticitySquared * sinPhi * sinPhi);
+            }
+
+            deltaLongitude = distance / localRad;
+            if (heading > 0.0) {
+                longitude = CesiumMath.negativePiToPi(start.longitude + deltaLongitude);
+            } else {
+                longitude = CesiumMath.negativePiToPi(start.longitude - deltaLongitude);
+            }
+        }
+
+        if (defined(result)) {
+            result.longitude = longitude;
+            result.latitude = latitude;
+            result.height = 0;
+
+            return result;
+        }
+
+        return new Cartographic(longitude, latitude, 0);
+    };
+
+    /**
+     * Provides the location of a point at the indicated longitude along the rhumb line.
+     * If the longitude is outside the range of start and end points, the first intersection with the longitude from the start point in the direction of the heading is returned. This follows the spiral property of a rhumb line.
+     *
+     * @param {Number} intersectionLongitude The longitude, in radians, at which to find the intersection point from the starting point using the heading.
+     * @param {Cartographic} [result] The object in which to store the result.
+     * @returns {Cartographic} The location of the intersection point along the rhumb line, undefined if there is no intersection or infinite intersections.
+     *
+     * @exception {DeveloperError} start and end must be set before calling function findIntersectionWithLongitude.
+     */
+    EllipsoidRhumbLine.prototype.findIntersectionWithLongitude = function(intersectionLongitude, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.number('intersectionLongitude', intersectionLongitude);
+        if (!defined(this._distance) || this._distance === 0.0) {
+            throw new DeveloperError('EllipsoidRhumbLine must have distinct start and end set.');
+        }
+        //>>includeEnd('debug');
+
+        var ellipticity = this._ellipticity;
+        var heading = this._heading;
+        var absHeading = Math.abs(heading);
+        var start = this._start;
+
+        intersectionLongitude = CesiumMath.negativePiToPi(intersectionLongitude);
+
+        if (!defined(result)) {
+            result = new Cartographic();
+        }
+
+        // If heading is -PI/2 or PI/2, this is an E-W rhumb line
+        // If heading is 0 or PI, this is an N-S rhumb line
+        if (Math.abs(CesiumMath.PI_OVER_TWO - absHeading) <= CesiumMath.EPSILON8) {
+            result.longitude = intersectionLongitude;
+            result.latitude = start.latitude;
+            result.height = 0;
+            return result;
+        } else if (CesiumMath.equalsEpsilon(Math.abs(CesiumMath.PI_OVER_TWO - absHeading), CesiumMath.PI_OVER_TWO, CesiumMath.EPSILON8)) {
```

Comment:
I assume these epsilons were from the ported code?  (if not, how did we choose them?) Same for the rest of the PR.

Reply:
Yes, it was chosen as the epsilon at which `Math.cos()` returns `1.0` similar to http://help.agi.com/AGIComponents/html/F_AGI_Foundation_Trig_AngleEpsilon.htm.